xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -4970,6 +4985,8 @@ void CVideoPlayer::GetVideoStreamInfo(int streamId, VideoStreamInfo &info)
   info.DestRect = s.DestRect;
   info.VideoRect = s.VideoRect;
   info.codecName = s.codec;
+  if (m_pInputStream->IsStreamType(DVDSTREAM_TYPE_DVD))
```

Comment:
Can this somehow be avoided by setting it on the struct (independently of the stream type) instead of the conditions when getting the info?

Reply:
I could use `m_CurrentStream.hint.dvd` but that particular test is used several times already in the code so I presumed it was the correct way to do it.

Reply:
my question was if we could set the conditions on the setter (wherever that is) and not on the getter. The getter should return whatever is on the struct without the need for checking any conditions

Reply:
Ah, right I see what you mean. If you omit the check, it will return `h262` instead of `mpeg-2`.  Maybe there is somewhere else in the code where h262 can be translated to mpeg2 but currently the GUI will not show a flag when fed with h262.  I don't know if that is purely a skin thing or somewhere in the GUI core code.  I do agree that if it could be handled elsewhere it would be better to do so.

Reply:
Ah if that is just a skin thing it should be fixed elsewhere. @jjd-uk might be able to help as he did similar things in the past

Reply:
IMO it's fine if it happens somewhere else as long as it's NOT in the skin itself....

Reply:
I only know about doing flagging in the skin, I'm mostly clueless when comes to core C++ stuff. However the only reference I see to h262 is https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamNavigator.cpp#L1605 so is that where it comes from? if so could that be change to mpeg2???

Reply:
Umm, the problem here is that no audio codec at all is returned without that check. I'll test to see if its possible to always use the hint value.  If so then the checks can be dropped.  Its not straightforward because there is no audio codec returned for a dvd stream without that check.

Reply:
that's all wrong here. Neither m_CurrentVideo/Audio nor m_pInputStream must be accessed from outside.